### PR TITLE
Update main 2.9 for main (Merge after CKAN redirected to maintenance page)

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

After CKAN has been redirected (https://github.com/alphagov/govuk-puppet/pull/11203) merge this branch in, most of the commits have already been reviewed on `main-2.9` branch, only commit - f2d3f77 onwards will there need to be some sort of review.

## Reference

https://trello.com/c/DZJFi8z5/2613-create-prs-to-prep-for-29-upgrade

### CODE REVIEW

File: bin/install-dependencies.sh [hunk 1]
```diff
@@ -18,6 +18,7 @@ pycsw_tag='2.4.0'
 pipopt='--exists-action=b'
 
 $pip install $pipopt -U numpy==1.16.4
+$pip install $pipopt -U setuptools_rust
```
Comment (ChrisBAshton) [inline bin/install-dependencies.sh]: Worth adding in the commit message why this is needed?

File: bin/install-dependencies.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ pycsw_tag='2.4.0'
 pipopt='--exists-action=b'
 
 $pip install $pipopt -U numpy==1.16.4
-$pip install $pipopt -U setuptools_rust
+$pip install $pipopt -U setuptools_rust pyOpenSSL
```
Comment (ChrisBAshton) [inline bin/install-dependencies.sh:21]: And here?

File: Jenkinsfile [hunk 3]
```diff
@@ -16,6 +16,7 @@ node ('!(ci-agent-4)') {
     stage('Installing Packages for python 3 release') {
       sh("rm -rf ./venv")
       sh("python3.6 -m venv ./venv")
+      sh("bash -c 'venv/bin/python -m pip install --upgrade 'pip==20.3.4''")
```
Comment (ChrisBAshton) [inline Jenkinsfile:19]: And here?

**Review by ChrisBAshton — APPROVED** [review]
LGTM, just a couple of comments about making the commit messages more descriptive.

**Comment by kentsanggds** [comment]
OK, I'll update the commit messages

File: bin/install-dependencies.sh (hunk 2 above)
Comment (kentsanggds) [reply bin/install-dependencies.sh]: decided to remove it as no longer needed after pip upgrade

File: bin/install-dependencies.sh (hunk 1 above)
Comment (kentsanggds) [reply bin/install-dependencies.sh]: decided to remove it as no longer needed after pip upgrade

File: Jenkinsfile (hunk 3 above)
Comment (kentsanggds) [reply Jenkinsfile]: Added a commit message to explain the need for an upgrade

**Pull request merged**